### PR TITLE
Java roadmap: jvm clarifications

### PR DESCRIPTION
**Pull request by leni-kirilov**
removed duplicate link
added JVM languages and GraalVM as a popular alternative

### CODE REVIEW

File: src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
 # How JVM Works
 
-The Java Virtual Machine is a program whose purpose is to execute other programs. JVMs are available for many hardware and software platforms (i.e. JVM is platform dependent). JVM is the one that actually calls the main method present in a java code. JVM is a part of JRE(Java Runtime Environment)
+The Java Virtual Machine (JVM) is a program that executes Java bytecode. JVMs are available for various hardware and software platforms, making them platform-dependent but the programs themselves - not. The so-called "Write once run everywhere" becomes true here. The JVM is responsible for calling the main method in Java code and is a crucial part of the Java Runtime Environment (JRE).
+
+While primarily designed for Java, the JVM can run any language that compiles to Java bytecode. This flexibility allows other languages like Kotlin, Scala, and Groovy to leverage the JVM ecosystem.
+
+Multiple JVM implementations exist, each with unique features. Oracle's HotSpot is the standard implementation, while GraalVM is a high-performance alternative that's gaining popularity. GraalVM offers enhanced performance.
```
Comment (dansholds) [inline src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md:7]: Can we try to condense all of this down into one paragraph. The new style guide we have for roadmap copy is less copy, more links 👍

File: src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 
 Visit the following resources to learn more:
 
-- [@article@What is JVM?](https://www.javatpoint.com/jvm-java-virtual-machine)
+- [@article@What is JVM & architectyre?](https://www.geeksforgeeks.org/jvm-works-jvm-architecture/)
```
Comment (dansholds) [inline src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md:11]: Typo in `architecture` & try to find another link other than the geeksforgeeks one, we already have 130+ links for that site so it's nice to add more diversity for our users 👍

**Review by dansholds — CHANGES_REQUESTED** [review]

File: src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md (hunk 2 above)
Comment (leni-kirilov) [reply src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md:11]: ok I'll look for another one, but I read it yesterday and it seemed good. Definitely better than the existing one which is pre Java8 and doesn't mention a lot of things

File: src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 Visit the following resources to learn more:
 
-- [@article@What is JVM?](https://www.javatpoint.com/jvm-java-virtual-machine)
+- [@article@What is JVM & architecture?](https://www.freecodecamp.org/news/jvm-tutorial-java-virtual-machine-architecture-explained-for-beginners/)
 - [@article@Introducing the Java Virtual Machine](https://www.infoworld.com/article/3272244/what-is-the-jvm-introducing-the-java-virtual-machine.html)
-- [@article@JVM (Java Virtual Machine) Architecture](https://www.javatpoint.com/jvm-java-virtual-machine)
 - [@video@How JVM works?](https://youtu.be/G1ubVOl9IBw)
 - [@feed@Explore top posts about JVM](https://app.daily.dev/tags/jvm?ref=roadmapsh)
+- [@article@JVM languages](https://www.whizlabs.com/blog/jvm-languages/)
+- [@article@ GraalVM](https://www.graalvm.org/)
```
Comment (dansholds) [inline src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md:12]:
```suggestion
- [@article@JVM languages](https://www.whizlabs.com/blog/jvm-languages/)
- [@article@ GraalVM](https://www.graalvm.org/)
- [@feed@Explore top posts about JVM](https://app.daily.dev/tags/jvm?ref=roadmapsh)
```

Comment (dansholds) [inline src/data/roadmaps/java/content/101-java-advanced-topics/106-how-jvm-works.md:12]:
```suggestion
- [@video@How JVM works?](https://youtu.be/G1ubVOl9IBw)
- [@article@JVM languages](https://www.whizlabs.com/blog/jvm-languages/)
- [@article@ GraalVM](https://www.graalvm.org/)
- [@feed@Explore top posts about JVM](https://app.daily.dev/tags/jvm?ref=roadmapsh)
```

**Review by dansholds — APPROVED** [review]
LGTM Now 👍

**Pull request merged**